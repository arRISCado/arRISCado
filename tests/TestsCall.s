	.option nopic
	.text
	.align	2
	.globl	main
	.type	main, @function
my_function:
    addi sp, sp, -4    
    sw ra, 0(sp)
    addi t0, zero, 5
    addi t1, zero, 1

    loop_start:
        beq t0, zero, loop_end
        add t1, t1, t0    
        addi t0, t0, -1 
        call loop_start        

    loop_end:
    
    add a0, zero, t1
    lw ra, 0(sp)   
    addi sp, sp, 4    
    jalr ra, 0(ra)

main:
    jal ra, my_function
    addi a2, zero, 10 
    
# Expected final state:
# t0: 0
# t1: 16
# a0: 16
# a2: 10
