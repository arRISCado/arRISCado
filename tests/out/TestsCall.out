0           x0	zero
x           x1	ra
x           x2	sp
x           x3	gp
x           x4	tp
0           x5	t0
16          x6	t1
x           x7	t2
x           x8	s0
x           x9	s1
16          x10	a0
x           x11	a1
10          x12	a2
x           x13	a3
x           x14	a4
x           x15	a5
x           x16	a6
x           x17	a7
x           x18	s2
x           x19	s3
x           x20	s4
x           x21	s5
x           x22	s6
x           x23	s7
x           x24	s8
x           x25	s9
x           x26	s10
x           x27	s11
x           x28	t3
x           x29	t4
x           x30	t5
x           x31	t6